src/lib/hookify/useBaseQuery.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  QueryClient,
+  QueryKey,
+  QueryObserver,
+  QueryObserverResult,
+  UseBaseQueryOptions,
+  notifyManager,
+} from '@tanstack/react-query';
+
+import { useCallbackHf, useEffectHf, useStateHf, useSyncExternalStoreHf } from './vanillaHooks';
+
+// This file is very similar to the one from the react-query codebase but changed to always use hookify hooks rather than react
+// sadly, this will probably need to be updated when we bump react-query major versions
```

Comment:
🙀

Reply:
Yes this is probably the saddest consequence of my choices, however note that this file is very short and only took like 30 minutes to put together.